src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -3572,36 +3572,47 @@ void func_80836BEC(Player* this, PlayState* play) {
         ignoreLeash = true;
     }
 
-    cond = func_8083224C(play);
-    if (cond || (this->unk_66C != 0) || (this->stateFlags1 & (PLAYER_STATE1_12 | PLAYER_STATE1_25))) {
-        if (!cond) {
+    isTalking = func_8083224C(play);
+
+    if (isTalking || (this->unk_66C != 0) || (this->stateFlags1 & (PLAYER_STATE1_12 | PLAYER_STATE1_25))) {
+        if (!isTalking) {
             if (!(this->stateFlags1 & PLAYER_STATE1_25) &&
                 ((this->heldItemAction != PLAYER_IA_FISHING_POLE) || (this->unk_860 == 0)) &&
                 CHECK_BTN_ALL(sControlInput->press.button, BTN_Z)) {
-
                 if (this->actor.category == ACTORCAT_PLAYER) {
-                    actorToTarget = play->actorCtx.attention.naviHoverActor;
+                    // The next lock-on actor defaults to the actor Navi is hovering over.
+                    // This may change to the arrow hover actor below.
+                    nextLockOnActor = play->actorCtx.attention.naviHoverActor;
                 } else {
-                    actorToTarget = &GET_PLAYER(play)->actor;
+                    // Dark Link will always lock onto the player.
+                    nextLockOnActor = &GET_PLAYER(play)->actor;
                 }
 
-                holdTarget = (gSaveContext.zTargetSetting != 0) || (this->actor.category != ACTORCAT_PLAYER);
+                // Get saved Z Target setting.
+                // Dark Link uses Hold Targeting.
+                isHoldTargeting = (gSaveContext.zTargetSetting != 0) || (this->actor.category != ACTORCAT_PLAYER);
+
                 this->stateFlags1 |= PLAYER_STATE1_15;
 
-                if ((actorToTarget != NULL) && !(actorToTarget->flags & ACTOR_FLAG_LOCK_ON_DISABLED)) {
-                    if ((actorToTarget == this->focusActor) && (this->actor.category == ACTORCAT_PLAYER)) {
-                        actorToTarget = play->actorCtx.attention.arrowHoverActor;
+                if ((nextLockOnActor != NULL) && !(nextLockOnActor->flags & ACTOR_FLAG_LOCK_ON_DISABLED)) {
+                    // Navi hovers over the current lock-on actor, so `nextLockOnActor` and `focusActor`
+                    // will be the same if already locked on.
+                    // In this case, `nextLockOnActor` will be the arrow hover actor instead.
+                    if ((nextLockOnActor == this->focusActor) && (this->actor.category == ACTORCAT_PLAYER)) {
+                        nextLockOnActor = play->actorCtx.attention.arrowHoverActor;
                     }
 
-                    if (actorToTarget != this->focusActor) {
-                        if (!holdTarget) {
+                    if (nextLockOnActor != this->focusActor) {
+                        // Set new lock-on
+                        if (!isHoldTargeting) {
```

Comment:
```suggestion
                        // Set new lock-on

                        if (!isHoldTargeting) {
```

Reply:
this suggestion confuses me

Reply:
I actually did it at first but wasnt sure.
The point is that the whole block is setting up the new lock on, not that if

Reply:
i see, should it just be moved above `if (nextLockOnActor != this->focusActor) {` then? all the other comments apply to the thing below them

Reply:
generally I put the comment above the conditions when talking about things in the context of the check. For example the above review discussion, im describing why `if (nextLockOnActor == this->focusActor)` is significant. Here i am just saying whats happening after the check